### DISCUSSION TITLE
difference between fit_transform() and fit() followed by transform() ?

### DISCUSSION THREAD

**Discussion by gabrielsevestre** [MaartenGr/BERTopic#2128] [Q&A]
Hello ! 

I am trying to extract topics from arXiv abstracts, but I am noticing very different behaviors with methods which I feel should give (more or less) the same results.

Indeed when calling fit_transform() on my data (with the BERTopic class), or when I fit and transform/predict with UMAP followed by HDBSCAN I get exactly the same clusters, but these are very different from when I call fit() and then transform() (still with the same BERTopic class). I get 5 times more outliers when transforming the same documents after applying the fit_transform() method !

However, this seems to not be caused by the inherent random initialization of HDBSCAN, because I get exactly the same results by method from different runs. I have set a random seed on the UMAP model. 

What could be the cause of this ?

**Comment by MaartenGr** [comment]
Thank you for sharing your question. There are a number of open issues/discussions on this very topic. The most recent one is [this one](https://github.com/MaartenGr/BERTopic/discussions/2124). You can find your answer there if I am not mistaken.
